plugin/trino-hive-hadoop2/src/test/java/io/trino/plugin/hive/AbstractTestHiveFileSystemS3.java

```diff
@@ -93,4 +132,137 @@ public void testIgnoreHadoopFolderMarker()
+        String basePrefix = basePath.toUri().getPath().substring(1);
+
+        // Expected file system tree:
+        // test-file-iterator-partitioned-listing-native-setup/
+        //      .hidden/
+        //          nested-file-in-hidden.txt
+        //      part=simple/
+        //          _hidden-file.txt
+        //          plain-file.txt
+        //      part=nested/
+        //          parent/
+        //             _nested-hidden-file.txt
+        //             nested-file.txt
+        //      part=plus+sign/
+        //          plus-file.txt
+        //      part=percent%sign/
+        //          percent-file.txt
+        //      part=url%20encoded/
+        //          url-encoded-file.txt
+        //      part=level1|level2/
+        //          pipe-file.txt
+        //          parent1/
+        //             parent2/
+        //                deeply-nested-file.txt
+        //      part=level1 | level2/
+        //          pipe-blanks-file.txt
+        //      empty-directory/
+        //      .hidden-in-base.txt
+
+        createFile(writableBucket, format("%s/.hidden/nested-file-in-hidden.txt", basePrefix));
+        createFile(writableBucket, format("%s/part=simple/_hidden-file.txt", basePrefix));
+        createFile(writableBucket, format("%s/part=simple/plain-file.txt", basePrefix));
+        createFile(writableBucket, format("%s/part=nested/parent/_nested-hidden-file.txt", basePrefix));
+        createFile(writableBucket, format("%s/part=nested/parent/nested-file.txt", basePrefix));
+        createFile(writableBucket, format("%s/part=plus+sign/plus-file.txt", basePrefix));
+        createFile(writableBucket, format("%s/part=percent%%sign/percent-file.txt", basePrefix));
+        createFile(writableBucket, format("%s/part=url%%20encoded/url-encoded-file.txt", basePrefix));
+        createFile(writableBucket, format("%s/part=level1|level2/pipe-file.txt", basePrefix));
+        createFile(writableBucket, format("%s/part=level1|level2/parent1/parent2/deeply-nested-file.txt", basePrefix));
+        createFile(writableBucket, format("%s/part=level1 | level2/pipe-blanks-file.txt", basePrefix));
+        createDirectory(writableBucket, format("%s/empty-directory/", basePrefix));
+        createFile(writableBucket, format("%s/.hidden-in-base.txt", basePrefix));
+
+        // List recursively through hive file iterator
+        HiveFileIterator recursiveIterator = new HiveFileIterator(
+                fakeTable,
+                Location.of(basePath.toString()),
+                trinoFileSystem,
+                new FileSystemDirectoryLister(),
+                new HdfsNamenodeStats(),
+                HiveFileIterator.NestedDirectoryPolicy.RECURSE);
+
+        List<String> recursiveListing = Streams.stream(recursiveIterator)
+                .map(TrinoFileStatus::getPath)
+                .toList();
+        // Should not include directories, or files underneath hidden directories
+        assertEqualsIgnoreOrder(
```

Comment:
`assertEqualsIgnoreOrder` exists in QueryAssertions class. Using the method for non-query result looks weird. Can we use AssertJ `assertThat` instead? Same for `AbstractTestHiveFileSystem`